core/vidl/vidl_ffmpeg_convert.cxx

```diff
@@ -27,9 +28,9 @@ extern "C" {
 }
 #endif
 
-
 //--------------------------------------------------------------------------------
 
+#if LIBAVCODEC_VERSION_MAJOR < 56
```

Comment:
There is a large block of code here that has been duplicated and modified for different versions.  It's hard to see what has changed, but it looks to be mostly the same.  Can you put the #if around just the parts that have changed to avoid so much code duplication?